src/OpenSSL/SSL.py

```diff
@@ -701,6 +711,35 @@ def set_default_verify_paths(self):
         """
         set_result = _lib.SSL_CTX_set_default_verify_paths(self._context)
         _openssl_assert(set_result == 1)
+        num = self._check_num_store_objects()
+        if num == 0:
+            self._fallback_default_verify_paths()
+
+    def _check_num_store_objects(self):
+        """
+        Checks how many objects are present in the Context's X509Store.
+
+        :return: int
+        """
+        store = self.get_cert_store()
+        sk_obj = _lib.X509_STORE_get0_objects(store._store)
+        _openssl_assert(sk_obj != _ffi.NULL)
+        return _lib.sk_X509_OBJECT_num(sk_obj)
+
+    def _fallback_default_verify_paths(self):
+        """
+        Default verify paths are based on the compiled version of OpenSSL.
+        However, when pyca/cryptography is compiled as a manylinux1 wheel
+        that compiled location can potentially be wrong. So, like Go, we
+        will try a predefined set of paths and attempt to load roots
+        from there.
+
+        :return: None
+        """
+        for cafile in _CERTIFICATE_FILE_LOCATIONS:
+            if os.path.isfile(cafile):
+                self.load_verify_locations(cafile)
```

Comment:
Welcome to the catch-22 of this approach.

If you throw an exception then users on, say, Gentoo, are going to be real mad that they can no longer use the system trust store just because they put some random data at `"/etc/pki/tls/cacert.pem"`, a path that has no special meaning on their system. If you *don't* throw an exception then you're silently swallowing potentially important errors. Maybe log?

Fun follow-on: should you load all certs at all these places? Does this open the user up to risks about having unexpected or malicious root certs elsewhere in the system suddenly be trusted by Python but nothing else? Doesn't this lead to PyOpenSSL behaving fundamentally differently to their native applications? Does this fundamentally mean that we should all abandon PyOpenSSL and just use PEP 543 instead? ARE WE NOT ALL DOOMED TO DIE ALONE?